compiler/src/intermediate_representation/constraint_bucket.rs

```diff
@@ -112,21 +114,72 @@ impl WriteLLVMIR for ConstraintBucket {
         const ASSERT_IDX: u32 = 0;
 
         match self {
-            ConstraintBucket::Substitution(_) => {
-                let lhs = get_instruction_arg(prev.into_instruction_value(), STORE_DST_IDX);
-                let rhs_ptr = get_instruction_arg(prev.into_instruction_value(), STORE_SRC_IDX);
-                let rhs = create_load(producer, rhs_ptr.into_pointer_value());
-                let constr = new_constraint(producer);
-                let call = create_call(
-                    producer,
-                    CONSTRAINT_VALUES_FN_NAME,
-                    &[
-                        to_basic_metadata_enum(lhs),
-                        to_basic_metadata_enum(rhs),
-                        to_basic_metadata_enum(constr),
-                    ],
-                );
-                Some(call)
+            ConstraintBucket::Substitution(i) => {
+                let size = match i.as_ref() {
+                    Instruction::Value(_) => todo!(),
```

Comment:
Not familiar enough to review this part of the coebase, but what are all these TODOs? If they are current limitations, please document them.